### PR TITLE
swap all odd and even bits

### PR DESCRIPTION
**Pull request by Ganesh-Agarwal-25**
The task is to swap all odd bits with even bits. For example, if the given number is 23 (00010111), it should be converted to 43(00101011). 
This PR is for Hacktober fest 2020.

# Welcome to Dart community

### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by Ganesh-Agarwal-25** [comment]
@Parowicz  please review and merge for Hacktober 2020.

File: other/swap_all_odd_and_even_bits.dart [hunk 1]
```diff
@@ -0,0 +1,9 @@
+//we are given an integer,the tast is to get its binary representation,swap all odd and even bits and print the new number
```
Comment (Parowicz) [inline other/swap_all_odd_and_even_bits.dart:1]: Could you add some tests?
```suggestion
import 'package:test/test.dart';

//we are given an integer,the tast is to get its binary representation,swap all odd and even bits and print the new number
```

File: other/swap_all_odd_and_even_bits.dart [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+int swapbits(int n) {
+  return (((n & 0xaaaaaaaa) >> 1) | ((n & 0x55555555) << 1));
+}
+
+void main() {
+  int n = 2;
+  print(swapbits(n));
```
Comment (Parowicz) [inline other/swap_all_odd_and_even_bits.dart:8]: For example
```suggestion
  test("swapbits returns 1 for 2", () {
    expect(swapbits(2), equals(1));
  });

  test("swapbits returns 23 for 43", () {
    expect(swapbits(43), equals(23));
  });

  test("swapbits returns 43 for 23", () {
    expect(swapbits(23), equals(43));
  });
```

**Comment by Ganesh-Agarwal-25** [comment]
@Parowicz I have applied the suggested commits,Thankyou.

**Review by Parowicz — APPROVED** [review]
LGTM!

**Comment by Ganesh-Agarwal-25** [comment]
@Parowicz thankyou so much, this is my first successful PR.

**Pull request merged**